ir/state.cpp

```diff
@@ -95,14 +126,27 @@ const StateValue& State::getAndAddUndefs(const Value &val) {
   return v;
 }
 
+const StateValue& State::getAndAddPoisonUB(const Value &val) {
+  auto &v = (*this)[val];
+  if (!v.non_poison.isTrue()) {
+    analysis.non_poison_vals.insert(&val);
+    if (v.non_poison.isBool())
+      addUB(v.non_poison);
+    else
+      // UB if any element is poison
+      addUB(v.non_poison != expr::mkUInt(-1, v.non_poison.bits()));
```

Comment:
uhm, i would make this == 0. We want all elements to be non-poison.

Reply:
Sorry, my mistake :/ Naively thought that non_poison being true means it should be 1 when it is a bitvector